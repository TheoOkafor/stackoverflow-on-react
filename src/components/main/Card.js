import React from 'react';
import PropTypes from 'prop-types';

const Card = (props) => {
  return (
    <div className={`card ${ props.styleName }`} id={ props.id } >
      {props.children}
    </div>
  );
}

Card.propTypes = {
  styleName: PropTypes.string,
  id: PropTypes.number,
  children: PropTypes.element,
}

export default Card;
