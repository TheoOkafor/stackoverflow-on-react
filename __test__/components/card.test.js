import React from 'react';
import Card from '../../src/components/main/Card';
import renderer from 'react-test-renderer';

test('Test that the Card displays', () => {
  const component = renderer.create(
    <Card styleName="primary">
      <p>Theo is just testing stuffs out</p>
    </Card>,
  );
  expect(component).toMatchSnapshot();
});
