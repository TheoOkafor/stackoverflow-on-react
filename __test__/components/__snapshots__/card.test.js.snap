// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Card displays 1`] = `
<div
  className="card primary"
>
  <p>
    Theo is just testing stuffs out
  </p>
</div>
`;
